src/graphics/lod_node.cpp

```diff
@@ -249,8 +242,43 @@ void LODNode::OnRegisterSceneNode()
     scene::ISceneNode::OnRegisterSceneNode();
 }
 
+void LODNode::autoComputeLevel(float scale)
+{
+    m_volume *= scale;
+
+    // Amount of details based on user's input
+    float agressivity = 1.0;
+    if(UserConfigParams::m_geometry_level == 0) agressivity = 1.0;
+    if(UserConfigParams::m_geometry_level == 1) agressivity = 0.75;
+    if(UserConfigParams::m_geometry_level == 2) agressivity = 0.5;
+
+    // First we try to estimate how far away we need to draw
+    float max_draw = 0.0;
+    max_draw = sqrtf((0.5 * m_volume + 10) * 200) - 10;
+    // If the draw distance is too big we artificially reduce it
+    if(max_draw > 250)
+    {
+        max_draw = 250 + (max_draw * 0.05);
```

Comment:
This formula will transform 251 into 250 + 0.05*251 = 263.55

You should tweak the formula to have a smooth transition.

For example : max_draw = 235 + (max_draw*0.06)